addons/ui/fnc_addItemContextMenuOption.sqf

```diff
@@ -0,0 +1,232 @@
+
+                                    MAGAZINE
+                                        RIFLE_MAGAZINE
+                                        RIFLE_MAGAZINE_GL
+                                        LAUNCHER_MAGAZINE
+                                        PISTOL_MAGAZINE
+
+                                    ASSIGNED_ITEM
+                                        MAP
+                                        GPS
+                                        RADIO
+                                        COMPASS
+                                        WATCH
+                                        HMD
+
+    _displayName            String keys are automatically translated. <STRING, ARRAY>
+        0: _displayName     - Option display name <STRING>
+        1: _tooltip         - Option tooltip <STRING>
+
+    _color                  - Option text color. Default alpha is 1.
+                              (default: [] = default color) <ARRAY>
+
+    _icon                   - Path to icon. (default: "" = no icon) <STRING>
+
+    _condition              <CODE, ARRAY>
+        0: _conditionEnable - Menu option is enabled and executed only if this
+                              condition reports 'true' (default: {true}) <CODE>
+        1: _conditionShow   - Menu option is shown only if this condition
+                              reports 'true'. (optional, default: {true}) <CODE>
+
+                            - Passed arguments:
+                              params ["_unit", "_container", "_item", "_slot", "_params"];
+
+    _statement              - Option statement (default: {}) <CODE>
+                              Return true to keep context menu opened.
+
+                            - Passed arguments:
+                              params ["_unit", "_container", "_item", "_slot", "_params"];
+
+    _consume                - Remove the item before executing the statement
+                              code. (default: false) <BOOLEAN>
+
+                            - This does NOT work for the following slots:
+                                GROUND
+                                CARGO
+
+    _params                 - Arguments passed as '_this select 4' to condition and
+                              statement (optional, default: []) <ANY>
+
+Returns:
+    Nothing/Undefined.
+
+Examples:
+    (begin example)
+        ["#All", "ALL", ">DEBUG ACTION<", nil, nil, {true}, {
+            params ["_unit", "_container", "_item", "_slot", "_params"];
+            systemChat str [name _unit, typeOf _container, _item, _slot, _params];
+            true
+        }, false, [0,1,2]] call CBA_fnc_addItemContextMenuOption;
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+
+// Force unscheduled environment to prevent race conditions.
+if (canSuspend) exitWith {
+    [CBA_fnc_addItemContextMenuOption, _this] call CBA_fnc_directCall;
+};
+
+if (!hasInterface) exitWith {};
```

Comment:
Swap with can suspend?

Reply:
I want to spent the least amount of time possible in scheduled environment.